eng/common/scripts/modules/ChangeLog-Operations.psm1

```diff
@@ -0,0 +1,109 @@
+# Common Changelog Operations
+
+$RELEASE_TITLE_REGEX = "(?<releaseNoteTitle>^\#+.*(?<version>\b\d+\.\d+\.\d+([^0-9\s][^\s:]+)?)\s(?<releaseStatus>\(Unreleased\)|\(\d{4}-\d{2}-\d{2}\))?)"
+
+# given a CHANGELOG.md file, extract the relevant info we need to decorate a release
+function Extract-ReleaseNotes {
+    param (
+        [Parameter(Mandatory = $true)]
+        [String]$ChangeLogLocation,
+        [String]$VersionString
+    )
+    
+    $ErrorActionPreference = 'Stop'
+
+    $releaseNotes = @{}
+    if ($ChangeLogLocation.Length -eq 0)
+    {
+        return $releaseNotes
+    }
+    
+    try 
+    {
+        $contents = Get-Content $ChangeLogLocation
+    
+        # walk the document, finding where the version specifiers are and creating lists
+        $version = ""
+        foreach($line in $contents){
+            if ($line -match $RELEASE_TITLE_REGEX)
+            {
+               if ($version -ne "")
+               {
+                  $releaseNotes[$version].ReleaseContent = $releaseContentBuilder -join [Environment]::NewLine
+                  $releaseNotes[$version].ReleaseNotesEntry = $releaseContentBuilder[1..($releaseContentBuilder.Length - 1)] -join [Environment]::NewLine
+               }
+               $version = $matches["version"]
+               $releaseContentBuilder = @()
+               $releaseNotes[$version] = New-Object PSObject -Property @{
+                  ReleaseVersion = $version
+                  ReleaseStatus = $matches["releaseStatus"]
+                  ReleaseContent = ""
+                  ReleaseNotesEntry = "" # Release content without the version title
+               }
+            }
+            $releaseContentBuilder += $line
+        }
+        $releaseNotes[$version].ReleaseContent = $releaseContentBuilder -join [Environment]::NewLine
```

Comment:
Once we change to incrementing each time through the loop you shouldn't need this.